Modules/Mod1A_functional_sim.py

```diff
@@ -2,66 +2,50 @@
 from ontobio.ontol_factory import OntologyFactory
 from ontobio.io.gafparser import GafParser
 from ontobio.assoc_factory import AssociationSetFactory
+from .generic_similarity import GenericSimilarity
 
+URI = 'http://geneontology.org/gene-associations/goa_{group}.gaf.gz'
+DEFAULT_GROUP = 'human'
 
+class FunctionalSimilarity(GenericSimilarity):
+    def __init__(self, ont='go', subject_category='gene', object_category='function', file=None, group=None):
+        if group != None:
+            file = URI.format(group=group)
 
-class FunctionalSim(object):
-    """
+        if file == None:
+            file = URI.format(group=DEFAULT_GROUP)
 
-    """
-    def __init__(self):
-        self.ontology = self.load_ontology(ont='go')
-        self.gene_set = ''
-        self.associations = ''
-        self.taxon = None
+        super(FunctionalSimilarity, self).__init__(
+            ont=ont,
+            subject_category=subject_category,
+            object_category=object_category,
+            file=file
+        )
 
-    def load_gene_set(self, gene_set, taxon=None):
-        self.gene_set = gene_set
-        self.taxon = taxon
+        self.symbol_map = {}
+        self.identifier_map = {}
 
-    def load_associations(self, group):
-        p = GafParser()
-        afactory = AssociationSetFactory()
-        url = "http://geneontology.org/gene-associations/gene_association.{}.gz".format(group)
-        if group == 'human':
-            url = "http://geneontology.org/gene-associations/goa_human.gaf.gz"
-        assocs = p.parse(url)
-        assocs = [x for x in assocs if 'header' not in x.keys()]
-        self.associations = afactory.create_from_assocs(assocs, ontology=self.ontology)
+    def load_gene_set(self, gene_set):
+        self.gene_set = []
 
-    def compute_similarity(self):
-        sims = list()
-        for gene in self.gene_set:
-            mg = QueryMyGene(curie=gene, taxon=self.taxon)
+        for gene in gene_set:
+            mg = QueryMyGene(curie=gene)
```

Comment:
We should consider using the official mygene python client in the future

Reply:
I forgot they have one when I did this

Reply:
@putmantime I just tested out the MyGene python client. It looks like it takes gene identifiers as input as opposed to CURIEs. 

Ex: NCBIGene:4750
```py
>>> import mygene
>>> mg = mygene.MyGeneInfo()
>>> mg.getgene('NCBIGene:4750') #no result
```

where as,
```py
>>> mg.getgene('4750')
{'_id': '4750', '_score': 13.696598, 'entrezgene': '4750', 'name': 'NIMA related kinase 1', 'symbol': 'NEK1', 'taxid': 9606}
```

gives proper results.

I see that the QueryMyGene wrapper does some CURIE transformation before querying MyGeneInfo. Do you recommend refactoring `QueryMyGene.query_mygene`  method to use the python client, leaving remaining transformation code intact?

Reply:
That's a good point, I did the transformation to make the API call, if we don't need more functionality than what is there, I think it's fine as is, but just wanted to make sure it was known that they do have a richly functional python client.

Reply:
It definitely makes sense to use the client instead. I'll refactor the `query_mygene` method